dotnet/src/InternalUtilities/src/Diagnostics/UnreachableException.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#if NETSTANDARD2_0
+
+// Polyfill for using UnreachableException with .NET Standard 2.0
```

Comment:
What is the advantage of using `UnreachableException` over `ArgumentOutOfRange` exception or `InvalidDataException`?

Reply:
I use UnreachableException in cases where it's like a debug assertion, i.e. expressing to the reader that if the throw line is reached, then there's a bug in the code itself. This is in contrast e.g. to incorrect user input, which may occur in a correct program - for those I use e.g. ArgumentOutOfRangeException.

So the difference is basically about giving more context to the person reading the code, i.e. conveying that the line should never ever be reached.